app/src/sharedTest/java/org/oppia/android/app/profile/ProfileChooserFragmentTest.kt

```diff
@@ -591,38 +650,280 @@ class ProfileChooserFragmentTest {
+    TestPlatformParameterModule.forceEnableOnboardingFlowV2(true)
+    profileTestHelper.initializeProfiles(autoLogIn = false)
+    launch(ProfileChooserActivity::class.java).use {
+      testCoroutineDispatchers.runCurrent()
+      scrollToPosition(position = 0)
+      verifyTextOnProfileListItemAtPosition(
+        itemPosition = 0,
+        targetView = R.id.profile_name_text,
+        stringToMatch = "Admin"
+      )
+      verifyTextOnProfileListItemAtPosition(
+        itemPosition = 0,
+        targetView = R.id.profile_is_admin_text,
+        stringToMatch = context.getString(R.string.profile_chooser_admin)
+      )
+      scrollToPosition(position = 1)
+      verifyTextOnProfileListItemAtPosition(
+        itemPosition = 1,
+        targetView = R.id.profile_name_text,
+        stringToMatch = "Ben"
+      )
+    }
+  }
+
+  @Test
+  fun testProfileChooserFragment_enableOnboardingV2_afterVisitingHomeActivity_showsJustNowText() {
+    TestPlatformParameterModule.forceEnableOnboardingFlowV2(true)
+    // Note that the auto-log in here is simulating HomeActivity having been visited before (i.e.
+    // that a profile was previously logged in).
+    profileTestHelper.initializeProfiles(autoLogIn = true)
+    launch<ProfileChooserActivity>(createProfileChooserActivityIntent()).use {
+      testCoroutineDispatchers.runCurrent()
+      onView(
+        atPositionOnView(
+          recyclerViewId = R.id.profiles_list,
+          position = 0,
+          targetViewId = R.id.profile_last_visited
+        )
+      ).check(matches(isDisplayed()))
+      verifyTextOnProfileListItemAtPosition(
+        itemPosition = 0,
+        targetView = R.id.profile_last_visited,
+        stringToMatch = "${context.getString(R.string.profile_last_used)} just now"
+      )
+    }
+  }
+
+  @Test
+  @Config(qualifiers = "land")
+  fun testFragment_enableOnboardingV2_landscapeMode_afterVisitingHome_showsJustNowText() {
+    TestPlatformParameterModule.forceEnableOnboardingFlowV2(true)
+    // Note that the auto-log in here is simulating HomeActivity having been visited before (i.e.
+    // that a profile was previously logged in).
+    profileTestHelper.addOnlyAdminProfile()
+    profileTestHelper.addMoreProfiles(8)
+    launch(ProfileChooserActivity::class.java).use {
+      testCoroutineDispatchers.runCurrent()
+      onView(
+        atPositionOnView(
+          recyclerViewId = R.id.profiles_list_landscape,
+          position = 0,
+          targetViewId = R.id.profile_last_visited
+        )
+      ).check(matches(isDisplayed()))
+      verifyTextOnProfileListItemAtPosition(
+        itemPosition = 0,
+        targetView = R.id.profile_last_visited,
+        stringToMatch = "${context.getString(R.string.profile_last_used)} just now",
+        recyclerViewId = R.id.profiles_list_landscape,
+      )
+    }
+  }
+
+  @Test
+  fun testFragment_enableOnboardingV2_addManyProfiles_checkProfilesSortedAndNoAddProfile() {
```

Comment:
Is the 'no add profile' bit actually being tested?

Also, I suggest moving that to its own test (so that verifying sorting & the add profile button absence are separate tests).

Reply:
Split the test into 2 and added the assertion for the add profile button.